packages/aws-cdk/lib/api/hotswap/ecs-services.ts

```diff
@@ -79,7 +127,14 @@ export async function isHotswappableEcsServiceChange(
               Labels: true,
             },
           },
-        });
+        } as const;
+        // We first uppercase the task definition to properly merge it with the one from CloudFormation template.
+        const upperCasedTaskDef = transformObjectKeys(target.taskDefinition, upperCaseFirstCharacter, excludeFromTransform);
+        // merge evaluatable diff from CloudFormation template.
+        const updatedTaskDef = applyPropertyUpdates(changes.updates, upperCasedTaskDef);
+        // lowercase the merged task definition to use it in AWS SDK.
+        const lowercasedTaskDef = transformObjectKeys(updatedTaskDef, lowerCaseFirstCharacter, excludeFromTransform);
```

Comment:
Wouldn't it be shorter to transform the updates to lowercase?

Reply:
I think no. There are certain keys that should not be transformed in a task definition, as specified in `excludeFromTransform`. If we transform the updates, it requires additional logic to determine if it's okay to transform the key or not because the schema is different in those extracted updates and we cannot reuse the `excludeFromTransform`.